engine/src/operators/OrderBy.cpp

```diff
@@ -321,9 +321,187 @@ void process_sort(blazing_frame & input, std::string query_part, Context * query
 	}
 }
 
+
+
+}  // namespace operators
+}  // namespace ral
+
+namespace ral {
+namespace operators {
+namespace experimental {
+
+	using Context = blazingdb::manager::experimental::Context;
+	using Node = blazingdb::transport::experimental::Node;
+	using namespace ral::distribution::experimental;
+
+std::unique_ptr<ral::frame::BlazingTable> process_sort(const ral::frame::BlazingTableView & table, std::string query_part, Context & context) {
+	auto rangeStart = query_part.find("(");
+	auto rangeEnd = query_part.rfind(")") - rangeStart - 1;
+	std::string combined_expression = query_part.substr(rangeStart + 1, rangeEnd);
+
+	std::string limitRowsStr = get_named_expression(combined_expression, "fetch");
+
+	size_t num_sort_columns = count_string_occurrence(combined_expression, "sort");
+
+	std::vector<int8_t> sortOrderTypes(num_sort_columns);
+	std::vector<int> sortColIndices(num_sort_columns);
+	for(int i = 0; i < num_sort_columns; i++) {
+		sortColIndices[i] = get_index(get_named_expression(combined_expression, "sort" + std::to_string(i)));
+		sortOrderTypes[i] =
+			(get_named_expression(combined_expression, "dir" + std::to_string(i)) == DESCENDING_ORDER_SORT_TEXT);		 
+	}
+
+	if(context.getTotalNodes() <= 1) {
+		if(num_sort_columns > 0) {
+			std::unique_ptr<ral::frame::BlazingTable> sorted_table = logicalSort(table, sortColIndices, sortOrderTypes);
+
+			if(!limitRowsStr.empty()) {
+				cudf::size_type limitRows = std::stoi(limitRowsStr);
+				if (limitRows < sorted_table->view().num_rows()){ // lets do the limit only if it will actually do sometihng
+					return logicalLimit(sorted_table->toBlazingTableView(), limitRows);
+				} else {
+					return sorted_table;
+				}
+			} else {
+				return sorted_table;
+			}
+		} else if(!limitRowsStr.empty()) {
+			cudf::size_type limitRows = std::stoi(limitRowsStr);
+			if (limitRows < table.view().num_rows()){ // lets do the limit only if it will actually do sometihng
+				return logicalLimit(table, limitRows);
+			} else {
+				return table.clone();
+			}
+		}
+	} else {
+		if(num_sort_columns > 0) {
+			return distributed_sort(context, table, sortColIndices, sortOrderTypes);
+		}
+		if(!limitRowsStr.empty()) {
+			// distributed_limit(context, input, std::stoi(limitRowsStr));
+		}
+	}
+}
+
+std::unique_ptr<ral::frame::BlazingTable>  distributed_sort(Context & context,
```

Comment:
What do you do if the data was sorted so you dont want to make a copy?

Reply:
then dont call the sort function

Reply:
then why are you checking that its sorted?

Reply:
i am not checking that its sorted. I dont follow. We dont really have a way for checking if its sorted.